### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by dimarogkov**

### CODE REVIEW

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,16 +1,70 @@
...
+      studentArr.sort((a: Student, b: Student) => (order === 'asc'
+        ? a.name.localeCompare(b.name)
+        : b.name.localeCompare(a.name)));
+      break;
+
+    case SortType.Surname:
+      studentArr.sort((a: Student, b: Student) => (order === 'asc'
+        ? a.surname.localeCompare(b.surname)
+        : b.surname.localeCompare(a.surname)));
+      break;
```
Comment (VitaliyBondarenko1982) [inline src/sortStudents.ts:41]:
```suggestion
    case SortType.Name:
    case SortType.Surname:
      studentArr.sort((a: Student, b: Student) => (order === 'asc'
        ? a[sortBy].localeCompare(b[sortBy])
        : b[sortBy].localeCompare(a[sortBy])));
      break;
```
this to cases with same logic

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,16 +1,70 @@
...
+          return 0;
+        }
+
+        if ((a.married && order === 'asc') || b.married) {
+          return 1;
+        }
+
+        return -1;
+      });
+      break;
```
Comment (VitaliyBondarenko1982) [inline src/sortStudents.ts:61]:
```suggestion
    case SortType.Age:
    case SortType.Married:
      studentArr.sort((a: Student, b: Student) => (order === 'asc'
        ? +a[sortBy] - +b[sortBy]
        : +b[sortBy] - +a[sortBy]));
      break;
```
same for age and married
converting true/false to number return 1 or 0

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]

File: src/sortStudents.ts (hunk 2 above)
Comment (dimarogkov) [reply src/sortStudents.ts:61]: I have some errors if I do like you
image
image

How I can fix this problem?

**Review by etojeDenys — CHANGES_REQUESTED** [review]
feel free to ask in chat

**Review by VitaliyBondarenko1982 — APPROVED** [review]

**Pull request closed without merging**